libraries/AP_Scheduler/AP_Scheduler.cpp

```diff
@@ -114,6 +121,10 @@ void AP_Scheduler::init(const AP_Scheduler::Task *tasks, uint8_t num_tasks, uint
     perf_info.set_loop_rate(get_loop_rate_hz());
     perf_info.reset();
 
+    if (_options & uint8_t(Options::RecordTaskInfo)) {
+        perf_info.allocate_task_info(num_tasks);
+    }
```

Comment:
I think add an else to free the task info if option not set. That will give a way to clear without rebooting. I often want to see the stats for only a particular flight mode for example

Reply:
Done!